corehq/apps/hqwebapp/views.py

```diff
@@ -1262,29 +1285,28 @@ def page_url(self):
 @require_superuser
 def create_alert(request):
     from corehq.apps.hqwebapp.models import MaintenanceAlert
+    import pytz
     alert_text = request.POST.get('alert_text')
-    domains = request.POST.get('domains').split() or None
-    MaintenanceAlert(active=False, text=alert_text, domains=domains).save()
-    return HttpResponseRedirect(reverse('alerts'))
-
-
-@require_POST
-@require_superuser
-def activate_alert(request):
-    from corehq.apps.hqwebapp.models import MaintenanceAlert
-    ma = MaintenanceAlert.objects.get(id=request.POST.get('alert_id'))
-    ma.active = True
-    ma.save()
-    return HttpResponseRedirect(reverse('alerts'))
+    domains = request.POST.get('domains')
+    domains = domains.split() if domains else None
 
+    start_time = request.POST.get('start_time')
+    end_time = request.POST.get('end_time')
+    timezone = request.POST.get('timezone') or 'UTC'
```

Comment:
Can you add in some validation (could be server or client side)?  Thinking specifically of:

* dates are in the future
* end_time is later than start_time

Reply:
@nospame this validation note is addressed with the added knockout stuff here https://github.com/dimagi/commcare-hq/pull/32415/commits/5975aed0829f380ede69746407aee4c50bc94e3c is that correct?

Reply:
@Robert-Costello Yes, though it doesn't ensure times will be in the future. However, Ethan and I talked offline and the fact that a date/time in the past would come up as "expired" seemed fine for an admin workflow.